[IMP] deploy: reword multi-thread/process section (#5176)
odoo/documentation: content/administration/install/deploy.rst

Builtin server
==============

> Odoo includes built-in HTTP, cron and live-chat servers, using either

Reviewer: ```suggestion
Odoo includes built-in HTTP, cron, and live-chat servers, using either
```

multi-threading or multi-processing.

> The **multi-threaded** server is a simpler server primarely used for
> development, demonstrations and for compatibility with various operation
> systems (including Windows). A new thread is spawned for every new HTTP
> request, even for long running requests such as longpolling. Extra daemonic

Reviewer: ```suggestion
The **multi-threaded** server is a simpler server primarily used for
development, demonstrations, and its compatibility with various operating
systems (including Windows). A new thread is spawned for every new HTTP
request, even for long-running requests such as long polling. Extra daemonic
```

cron threads are spawned too. Due to a Python limitation (GIL), it doesn't make
the best use of the hardware.

> The multi-threaded server is the default server, also for docker containers, it
> is selected by leaving the :option:`--workers <odoo-bin --workers>` option out
> or to set it to ``0``.

Reviewer: ```suggestion
The multi-threaded server is the default server. For docker containers, it
is selected by leaving the :option:`--workers <odoo-bin --workers>` option out
or setting it to ``0``.
```

Author: > The multi-threaded server is the default server, also for docker containers. It
is selected by leaving the :option:`--workers <odoo-bin --workers>` option out
or setting it to ``0``.

The **multi-processing** server is a full blown server primarely used for
production. It is not liable to the same Python limitation (GIL) on resource
usage hence makes best use of the hardware. A pool of workers is created upon
server startup, new HTTP requests are queued by the OS until there are workers
ready to process them. An extra event driven HTTP worker for the live-chat is
spawned on an alternative port. Extra cron workers are spawned too. A
configurable process reaper monitores resource usage and is able to
kill/restart failed workers.

> The multi-processing server is opt-in, it is selected by setting

Reviewer: ```suggestion
The multi-processing server is opt-in. It is selected by setting
```

LiveChat
--------

> In multi-processing, a dedicated LiveChat worker is automatically started and listens on the :option:`--longpolling-port <odoo-bin --longpolling-port>`. By default, the HTTP requests will keep accessing the normal HTTP workers instead of the LiveChat one, you must deploy a proxy in front of Odoo and redirect incoming requests whose path starts with ``/longpolling`` to the LiveChat worker. You must also start Odoo in :option:`--proxy-mode <odoo-bin --proxy-mode>` so it uses the real client headers (such as hostname, scheme, and IP) instead of the proxy ones.

Reviewer 2: That's a veeery long line :wink:

Author: fixed, I also made sure that all other lines where wrap at 100 chars (documentation guidelines)

Cron Workers
------------

Starting one of the built-in Odoo servers next to the WSGI server is required to
process cron jobs. That server must be configured to only process crons and not
HTTP requests using the :option:`--no-http <odoo-bin --no-http>` cli option or
the ``http_enable = False`` configuration file setting.

> On Linux like systems, it is recommended to use the multi-processing server over
> the multi-threading one to benefit from the better hardware usage and increased
> stability, i.e. using the :option:`--workers=-1 <odoo-bin --workers>`

Reviewer: ```suggestion
On Linux-like systems, using the multi-processing server over
the multi-threading one is recommended to benefit from better hardware usage and increased
stability, i.e., using the :option:`--workers=-1 <odoo-bin --workers>`
```

and :option:`--max-cron-threads=n <odoo-bin --max-cron-threads>` cli options.

LiveChat
--------

Using a gevent-compatible WSGI server is required for the correct operation of
the live chat feature. That server should be able to handle many simultaneous
long-lived HTTP requests but doesn't need a lot of processing power. All
requests whose path starts with ``/longpolling`` should be directed to that
server. A regular (thread/process-based) WSGI server should be used for all
other requests.

> Reusing the Odoo cron server to serve the live chat requests is possible.

Reviewer 3: I'm not sure that `to reuse` is the appropriate verb to use here.
As it means `to use again especially in a different way`.
But here the cron server is still used as a cron server **and** as a longpolling one.
Why not something like
```suggestion
The Odoo cron server can also be used to serve the live chat requests.
```

Author: suggestion applied

> It is possible to reuse the Odoo cron server to serve the live chat requests.

Reviewer: ```suggestion
Reusing the Odoo cron server to serve the live chat requests is possible. 
```
